File: Factorial/C++/Factorial.cpp
#include <algorithm>
#include <string>
#include <iostream>

long long int factorial(long long int n)
{
   long long int r = 1;
   while(1 < n) {
       r *= n--;
   }
   return r;
}

int main( ) {
   int number = 10;
   std::cout << factorial(number) << std::endl;
   return 0 ;
}

File: Factorial/Java/Factorial.java
import java.math.BigInteger;
import java.util.InputMismatchException;
import java.util.Scanner;

public class Factorial {

  public BigInteger factorial(BigInteger n) {
    if ( n == null ) {
      throw new IllegalArgumentException();
    }

    else if ( n.equals(BigInteger.ZERO) ) {
      return BigInteger.ONE;
    }
    else if ( n.signum() == - 1 ) {
      // negative
      throw new IllegalArgumentException("Argument must be a non-negative integer");
    }
    else {
      return n.equals(BigInteger.ONE)
          ? BigInteger.ONE
          : factorial(n.subtract(BigInteger.ONE)).multiply(n);
    }
  }

  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    BigInteger number, result;
    number = new BigInteger("100");
    result = new Factorial().factorial(number);
    System.out.println("Factorial of " + number + ": " + result);

  }

}